### ISSUE TITLE
[Module Proposal]: `Logic Apps Standard`

### ISSUE THREAD

**Issue by scottroot-msft** [Azure/Azure-Verified-Modules#647]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Bicep

### Module Classification?

Resource Module

### Module Name

avm/res/website/logic-app-standard

### Module Details

I discussed this with Lohith and some other folks from the AVM chat we had yesterday.

Currently there is a logic apps module (consumption).  There is no specific module for Logic Apps standard.

I am told to use the Website module and specify the 'kind' parameters to include workflowapp etc. accordingly.  Not a problem.

In discussing this with Lohith, he wanted me to submit this as an issue for community discussion to see if it made sense to have a module that was just 'preconfigured' with the kind attribute or not.  There also is no example for logic app standard on the website module, so that could be an option as well.

I know that there is a dependency on an app service plan, and that for logic apps only the WS* plans are supported, so this may also be worth discussing.

Thanks for considering logic apps standard.

### Do you want to be the owner of this module?

No

**Comment by lohithgn** [comment]
@matebarabas @PmeshramPM can you folks check this one. As @scottroot-msft  said - i like the idea of having a dedicated resource module called Logic App Standard. All the nuances of how this to be set up (i.e. this is a website underneath, with a specific plan which is of type WorkflowStandard-1/2/3) can be in this resource module. We i.e. MS have been asking cx to move to LA Standard from LA ISE - coz ISE is getting retired on Aug 2024.

thoughts?

**Comment by matebarabas** [comment]
Based on our internal discussion, @lohithgn will further clarify this proposal with @scottroot-msft and will assess if this request is best positioned as a [feature request](https://aka.ms/AVM/Bicep/ModuleIssue) for the existing module of `avm/res/web/site` (in which case this issue will be closed), or as a pattern module (in which case this issue will need to be updated with a new the module name and pattern module classification).

**Comment by scottroot-msft** [comment]
> Has the discussion been held? I would be interested in either working on a new resource or helping to extend the existing - happy to discuss internally if needed.

 @lohithgn and I did discuss a while back and it was communicated to me that a logic apps standard AVM would be better suited in the patterns module.  Is this still the case @lohithgn ?

**Comment by lohithgn** [comment]
@scottroot-msft Yes. If we need lets say a logic-app module - it will have to be a pattern module. 
@matebarabas what needs to be done on this issue to turn this request to a pattern module request?

**Comment by matebarabas** [comment]
@lohithgn, @scottroot-msft, I've just reached out to you internally with some clarifying questions.

**Comment by DieterGobeyn** [comment]
Hi all,

I can confirm that the site module is indeed cumbersome and not ideally suited for the mandatory configuration of Logic App Standard. To make it work, which adds unnecessary complexity.

Could you clarify the direction for this proposal? Are you planning to develop and release a dedicated Logic App Standard module, or do you intend to enhance the existing site module with specific Logic App functionality? A clear roadmap for this would be highly beneficial.

Thank you!

**Comment by Nepomuceno** [comment]
THere has been no updated in this module is this still active ?

**Comment by matebarabas** [comment]
Hi @Nepomuceno, there were multiple internal discussions to cover the idea of this Bicep module. The conclusion was that the related resource module (`avm/res/web/site`) needs to be made compatible with Logic App Standard (it needs to offer the required sku/tier), and then a pattern module could be developed for it. Once someone wants to build and own that pattern module, this proposal can be converted to cover for it.
